### PR TITLE
Set `-F` in `OTHER_SWIFT_FLAGS`

### PR DESCRIPTION
**Pull request by chiragramani**
This PR addresses the following:
1. suggestions from the framework should be shown in the IDE.
2. Tapping on the framework name in the import statement, and jump to definition should open the framework interface.

Steps to repro:
1. Download the project from [repro_framework.zip](https://github.com/MobileNativeFoundation/rules_xcodeproj/files/11619131/repro_framework.zip)
2. 
```
bazel run :xcodeproj
```
3. Type AEAD in the swift file and auto-completion should work as expected.
4. Tapping on `import CryptoSwift` should not error out and should open the framework interface file.

### CODE REVIEW

**Comment by brentleyjones** [comment]
Can you adjust the tests similar to the last PR as well?

**Comment by chiragramani** [comment]
> Can you adjust the tests similar to the last PR as well?

absolutely, on it!

File: examples/integration/test/fixtures/bwb.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -11674,8 +11674,8 @@
...
+				OTHER_SWIFT_FLAGS = "-Xcc -working-directory -Xcc $(PROJECT_DIR) -working-directory $(PROJECT_DIR) -Xcc -ivfsoverlay -Xcc $(OBJROOT)/bazel-out-overlay.yaml -vfsoverlay $(OBJROOT)/bazel-out-overlay.yaml @$(DERIVED_FILE_DIR)/swift.compile.params -I$(CURRENT_EXECUTION_ROOT)/bazel-out/watchos-x86_64-min7.0-applebin_watchos-watchos_x86_64-dbg-STABLE-2/bin/Lib -F$(CURRENT_EXECUTION_ROOT)/external/com_github_krzyzanowskim_cryptoswift/CryptoSwift.xcframework/watchos-arm64_i386_x86_64-simulator";
+				"OTHER_SWIFT_FLAGS[sdk=watchos*]" = "-Xcc -working-directory -Xcc $(PROJECT_DIR) -working-directory $(PROJECT_DIR) -Xcc -ivfsoverlay -Xcc $(OBJROOT)/bazel-out-overlay.yaml -vfsoverlay $(OBJROOT)/bazel-out-overlay.yaml @$(DERIVED_FILE_DIR)/swift.compile.params -I$(CURRENT_EXECUTION_ROOT)/bazel-out/watchos-arm64_32-min7.0-applebin_watchos-watchos_arm64_32-dbg-STABLE-5/bin/Lib -F$(CURRENT_EXECUTION_ROOT)/external/com_github_krzyzanowskim_cryptoswift/CryptoSwift.xcframework/watchos-arm64_32_armv7k";
```
Comment (brentleyjones) [inline examples/integration/test/fixtures/bwb.xcodeproj/project.pbxproj:11678]: Note to self: we need to adjust the prefixes (in a follow up PR) to use `$(BAZEL_EXTERNAL)` and the like. This makes it resilient to bazel messing with the symlinks.

File: test/internal/opts/process_compiler_opts_tests.bzl [hunk 2]
```diff
@@ -149,8 +149,6 @@ def process_compiler_opts_test_suite(name):
             "arm64-apple-ios15.0-simulator",
             "-sdk",
             "__BAZEL_XCODE_SDKROOT__",
-            "-F__BAZEL_XCODE_DEVELOPER_DIR__/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks",
-            "-F__BAZEL_XCODE_SDKROOT__/Developer/Library/Frameworks",
```
Comment (brentleyjones) [inline test/internal/opts/process_compiler_opts_tests.bzl:153]: Can you add `-F` to the test that is testing `-F`? In all the same variations?

File: tools/params_processors/swift_compiler_params_processor.py [hunk 3]
```diff
@@ -187,7 +187,7 @@ def _inner_process_swiftcopts(
         is_bwx: bool) -> Optional[str]:
     is_clang_opt = opt == "-Xcc" or previous_opt == "-Xcc"
 
-    if not is_clang_opt and (previous_opt == "-I" or opt.startswith("-I")):
+    if not is_clang_opt and (previous_opt == "-I" or opt.startswith("-I") or previous_opt == "-F" or opt.startswith("-F")):
```
Comment (brentleyjones) [inline tools/params_processors/swift_compiler_params_processor.py:190]: Can you wrap this at 80 characters?

File: tools/params_processors/swift_compiler_params_processor_tests.py [hunk 4]
```diff
@@ -448,7 +448,6 @@ def _parse_args(args):
                 build_mode = "bazel",
             ),
             [
-                "'-F$(DEVELOPER_DIR)/Hi'",
```
Comment (brentleyjones) [inline tools/params_processors/swift_compiler_params_processor_tests.py:451]: Can you add `-F` to the test that is testing `-I`, in all the same variations?

Comment (chiragramani) [reply tools/params_processors/swift_compiler_params_processor_tests.py:451]: Thanks, updated!

File: test/internal/opts/process_compiler_opts_tests.bzl (hunk 2 above)
Comment (chiragramani) [reply test/internal/opts/process_compiler_opts_tests.bzl:153]: for sure, updated!

**Review by mattrobmattrob — APPROVED** [review]

**Pull request merged**